e2e: Set security context for all the applications

Madhu-1 opened the pull request:
Some applications are still using the root user which is not good to have, setting the security context for all the applications to follow the kubernetes security suggestions and same will be tested in the CI.

Madhu-1 committed:
e2e: run pod with less privilege

Set the default securityContext
to allow pod to run as non root
and with default permissions.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
deploy: set securityContext for all applications

setting the securityContext for all the application
we have in the repo, Using centos image as nginx
image is built with root user.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
/test ci/centos/mini-e2e-helm/k8s-1.33

Madhu-1 committed:
ci: log pod message and reason

If the pod is not in the Running
state, log the message and reason
to get more details.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
/test ci/centos/mini-e2e-helm/k8s-1.33

Madhu-1 commented:
/test ci/centos/mini-e2e-helm/k8s-1.33

Madhu-1 commented:
/test ci/centos/mini-e2e-helm/k8s-1.33

Madhu-1 committed:
ci: fix readonly error message

The error message doesnt contains
"cannot create" as it depends on
the container image. Updating
the error message to work for all
container images.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
/test ci/centos/mini-e2e-helm/k8s-1.33

Madhu-1 commented:
/test ci/centos/mini-e2e-helm/k8s-1.32

Madhu-1 committed:
e2e: increase e2e timeout

it looks like e2e is getting timeout
increasing the test timeout.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
Adding one more commit to increase the e2e timeout.

Madhu-1 commented:
/test ci/centos/mini-e2e-helm/k8s-1.32

nixpanic commented:
I just checked this too, it looks like the timeout hits, no other unexpected delays. Do you have a theory why this PR makes the CI jobs (a little) slower?

Madhu-1 commented:
> I just checked this too, it looks like the timeout hits, no other unexpected delays. Do you have a theory why this PR makes the CI jobs (a little) slower?

Not yet, i need to see why now its taking more time. Even though the image policy is Pull if not present.

Madhu-1 commented:
still CI is getting timeout, I need to spend time and analyze if there is any extra time spent on chown/chmod etc due to the user change or even something else. I will keep it posted

Madhu-1 committed:
e2e: impore pod debug logging

improving the app creation and
deletion debug logging so that
it will help us to calculate the
time taken for pod creation and
deletion.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
/test ci/centos/mini-e2e-helm/k8s-1.32

Madhu-1 commented:
Pod creation/deletion is taking more time, added logs to verify it. adding label until am sure where more time is taken

Madhu-1 commented:
## With normal user (with this PR)
```
Total time spent on pod creation: 28 minutes and 34 seconds
Total time spent on pod delete: 96 minutes and 46 seconds
```

## With Root user (without this PR)
```
Total time spent on pod creation: 30 minutes and 18 seconds
Total time spent on pod delete: 22 minutes and 54 seconds
```

The pod deletion is taking more time. because of that the tests are timing out

Madhu-1 commented:
Closing this one for now as making all the pod Users as normal adds more time to E2E testing.
